python/ray/dataframe/dataframe.py

```diff
@@ -711,7 +712,72 @@ def dropna(self, axis, how, thresh=None, subset=[], inplace=False):
             If inplace is set to True, returns None, otherwise returns a new
             DataFrame with the dropna applied.
         """
-        raise NotImplementedError("Not yet")
+        if is_list_like(axis):
+            raise NotImplementedError(
+                "To contribute to Pandas on Ray, please visit "
+                "github.com/ray-project/ray.")
+
+        axis = pd.DataFrame()._get_axis_number(axis)
+        inplace = validate_bool_kwarg(inplace, "inplace")
+
+        if axis == 1 and subset is not None:
+            subset = self.index.isin(subset)
```

Comment:
```python
subset = [item for item in self.index if item in subset]
```
Is slightly faster. Also you can probably refactor a bit by putting the top level if statement just be `if subset is not None`